src/doctor/checks/ios.ts

```diff
@@ -424,25 +437,64 @@ async function extractPodVersions(project: iOSProject): Promise<void> {
   const podfileLock = project.podfileLock;
   const podfileLockContent = podfileLock.content;
 
-  const validatePod = (podName: string, optional: boolean = false): boolean => {
+  const validatePod = (
+    podName: string,
+    optional: boolean = false,
+    // Minimum required version for new features
+    minRequiredVersion: string | undefined = undefined,
+    // Message to display when the pod needs to be updated for new features
+    updatePodMessage: string = `Please update ${podName} to latest version following our documentation`
+  ): boolean => {
     let podVersions: string | undefined;
     if (podfileLockContent) {
       podVersions = extractVersionFromPodLock(podfileLockContent, podName);
     }
 
     if (podVersions) {
       logger.success(`${podName}: ${podVersions}`);
+
+      // Check if the pod version is below the minimum required version
+      // If so, alert the user to update the pod for new features
+      if (
+        minRequiredVersion &&
+        compareSemanticVersions(
+          extractSemanticVersion(podVersions),
+          minRequiredVersion
+        ) < 0
+      ) {
+        logger.alert(updatePodMessage);
+      }
     } else if (!optional) {
       logger.failure(`${podName} module not found`);
     }
     return podVersions !== undefined;
   };
 
-  validatePod(POD_TRACKING);
+  if (project.framework == 'iOS') {
+    // Since iOS SDK 3.0.0, we have removed Tracking module and apps should use Data Pipeline module now
+    // Validate Data Pipeline module for native iOS apps
+    validatePod(POD_DATA_PIPELINE);
```

Comment:
maybe not in this phase but we should check if the data pipeline pod doesn't exist then we check for tracking. Also if both exist then there is a problem.

Reply:
Yeah, good check to add when we work on full support for V3 changes.